crates/edr_napi/index.d.ts

```diff
@@ -804,6 +821,86 @@ export interface BaseCounterExample {
   /** See [edr_solidity_tests::fuzz::BaseCounterExample::args] */
   readonly args?: string
 }
+/**
+ * Object representing a call in an execution trace, including contract
+ * creation.
+ */
+export interface CallTrace {
+  /** The kind of call or contract creation this represents. */
+  kind: CallKind
+  /** Whether the call succeeded or reverted. */
+  success: boolean
+  /** Whether the call is a cheatcode. */
+  cheatcode: boolean
+  /** The amount of gas that was consumed. */
+  gasUsed: bigint
+  /** The amount of native token that was included with the call. */
+  value: bigint
+  /**
+   * The target of the call. Provided as a contract name if known, otherwise
+   * a checksum address.
+   */
+  contract: string
+  /**
+   * The input (calldata) to the call. If it encodes a known function call,
+   * it will be decoded into the function name and a list of arguments.
+   * For example, `{ name: "ownerOf", arguments: ["1"] }`. Note that the
+   * function name may also be any of the special `fallback` and `receive`
+   * functions. Otherwise, it will be provided as a raw byte array.
+   */
+  inputs: DecodedTraceParameters | Uint8Array
+  /**
+   * The output of the call. This will be a decoded value if the function is
+   * known, otherwise a raw byte array.
+   */
+  outputs: string | Uint8Array
+  /**
+   * Interleaved subcalls and event logs. Use `kind` to check if each member
+   * of the array is a call or log trace.
+   */
+  children: Array<CallTrace | LogTrace>
```

Comment:
Is there a danger in using `CallKind | LogKInd` due to the way that enums are represented in JS? E.g. what if both types have a variant in common?

Reply:
Oh, I see what you're saying! Yes, `CallKind.Call` and `LogKind.Log` have the same numeric runtime representation, and `trace.kind === LogKind.Log` doesn't correctly narrow the type of `trace`!

One fix would be to assign explicit numeric values:
```rust
pub enum CallKind {
    Call = 0,
    CallCode = 1,
    DelegateCall = 2,
    StaticCall = 3,
    Create = 4,
}

pub enum LogKind {
    Log = 5,
}
```

Another possible fix would be to define a single enum and use a `#[napi(ts_type = ...)]` annotation to export a narrower type. The result of this would be closer to what one would do in TypeScript directly.

Reply:
Fixed using the first option: 3f5fb9b34a5b

I'd love to have some sort of test or static assertion for this but I couldn't find any simple way to do it.